src/server/user-controller.js

```diff
@@ -111,10 +112,16 @@ class UserController {
     return this.pageRenderer.sendPage(req, res, PAGE_NAME.login, {});
   }
 
-  handleGetLogoutPage(req, res) {
+  async handleGetLogoutPage(req, res, next) {
     if (req.isAuthenticated()) {
-      req.logout();
-      res.clearCookie(this.serverConfig.sessionCookieName);
+      const logout = util.promisify(req.logout);
+
+      try {
+        await logout();
+        res.clearCookie(this.serverConfig.sessionCookieName);
+      } catch (err) {
+        return next(err);
+      }
```

Comment:
All the handlers in educandu are automatically wrapped into a promise calling `next` in case of an error (done by the `express-async-errors` package.

So it's enough to change it like (we should actually never call `next` in a handler, only in middlewares):

~~~
  async handleGetLogoutPage(req, res) {
    if (req.isAuthenticated()) {
      await util.promisify(req.logout)();
      res.clearCookie(this.serverConfig.sessionCookieName);
    }

    return res.redirect(routes.getDefaultLogoutRedirectUrl());
  }
~~~

Looks like `passport.js` has changed the `logout` function from synchronous to asynchronous and therefore demands a callback now (probably not really documented clearly).

Reply:
Interesting. Will reduce the `next` call, thanks 👍 Yeah, by "documented clearly" you mean "not documented", right? :D